components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -521,6 +523,30 @@ public KeyManagerConfigurationDTO addKeyManagerConfiguration(
                     "Key manager Already Exist by Name " + keyManagerConfigurationDTO.getName() + " in tenant " +
                             keyManagerConfigurationDTO.getOrganization(), ExceptionCodes.KEY_MANAGER_ALREADY_EXIST);
         }
+
+        if (keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY) != null) {
+            try {
+                IdentityProvider idp = IdentityProviderManager.getInstance().getIdPByMetadataProperty(
+                        IDP_ISSUER_NAME,
+                        keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY).toString(),
+                        APIUtil.getInternalOrganizationDomain(keyManagerConfigurationDTO.getOrganization()),
+                        false);
+                if (idp != null) {
+                    throw new APIManagementException(
+                            String.format("Key Manager with issuer %s is already registered in tenant %s",
+                                    keyManagerConfigurationDTO.getAdditionalProperties().get(ISSUER_KEY).toString(),
+                                    keyManagerConfigurationDTO.getOrganization()), ExceptionCodes.DUPLICATE_ISSUER);
+                }
+            } catch (IdentityProviderManagementException e) {
+                throw new APIManagementException(String.format("IdP adding failed. %s", e.getMessage()),
+                        e,
+                        ExceptionCodes.IDP_ADDING_FAILED);
```

Comment:
This is most probably an Internal Server Error right. But IDP_ADDING_FAILED returning 400 may not give the correct information. Do we know what other places IDP_ADDING_FAILED is used and based on that is it okay to change its status code to 500?

Reply:
IDP_ADDING_FAILED is used in two places. 

1. Add IDP https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java#L516-L543
2. Edit IDP https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java#L677-L712

Both are thrown catching the IdentityProviderManagementException. I think both scenarios have similar behavior.

Reply:
Changed the error codes with https://github.com/wso2/carbon-apimgt/pull/11118/commits/c7402c0dac7b184bd56abb1513f4cad797ce0917